test/integration/datasources_k8s.bats

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env bats
+
+load helper
+
+function setup () {
+  mkdir -p /var/run/secrets/kubernetes.io/serviceaccount
+  touch /var/run/secrets/kubernetes.io/serviceaccount/token
+  cp /tmp/ca.crt  /var/run/secrets/kubernetes.io/serviceaccount/
+  start_k8s_svc
+  export KUBERNETES_SERVICE_PORT=8083 KUBERNETES_SERVICE_HOST=127.0.0.1
+
+}
+
+function teardown () {
+  stop_k8s_svc
+  rm -rf /var/run/secrets/kubernetes.io
+}
+
+@test "Testing k8s" {
+  gomplate -i '{{k8s.ConfigMap "default" "test" "test.property.1" "default"}}'
```

Comment:
This should look more like this:

```bash
gomplate -d k8s=k8s:// -i '{{(datasource "k8s" "default" "test" "test.property.1" "default" )}}' 
```

(though it probably makes sense for the syntax to be quite a bit less complex)

Reply:
The point is that to get a configmap we need 3 parameters, the k8s namespace, to configmap name and the key we want to retrieve. About the  "-d k8s=k8s://" I can add it, but when inside a k8s the k8s client know how to connect, so there will be not any kind of uri there.

Reply:
The point is, the k8s configmap is more similar to the aws metadata, that to a datasource.

Reply:
Hrm... Would a URL like `k8s:///namespace/configmap_name/key` make sense? This would be similar to the way the `vault://` URI scheme works.

You have a good point w.r.t. the `aws` functions. I did have a vague plan _at some point_ to make an AWS datasource, but that's more complicated because of the way their various APIs are arranged.

The advantage with using datasources is that it abstracts the source from the template. I could render the same template using a JSON file as a datasource in one scenario, or a Kubernetes ConfigMap in another scenario.

_For now_, let's please start with the datasource approach. If it becomes clear that a `k8s` namespace is also necessary, we can add it on afterwards.